assignment2.py

```diff
@@ -0,0 +1,155 @@
+class ListNode:
+    def __init__(self, value):
+        self.next = None
+        self.value = value
+        
+def string_compression(s):
+    count = 1
+
+    compressed = []
+
+    for i in range(1, len(s)):
+        if s[i - 1] == s[i] :
+            count += 1
+
+        else:
+            compressed.append(s[i - 1])
+            compressed.append(str(count))
+            #reset count
+            count = 1
+
+    compressed.append(s[-1])
+    #append conpressed amt
+    compressed.append(str(count))
+
+    compressed_str = ''.join(compressed)
+    #checkcase for if its og
+    return compressed_str if len(compressed_str) < len(s) else s
+
+def add_linked_lists_reverse(l1, l2):
+    carry = 0
+
+    dummy_head = ListNode(0)
+    current = dummy_head
+
+    while l1 or l2:
+        x = l1.value if l1 else 0
+        y = l2.value if l2 else 0
+
+        total = x + y + carry
+        carry = total // 10
+        current.next = ListNode(total % 10)
```

Comment:
This may not be useful since you're already passing `total % 10` directly to the `ListNode` constructor, but Python has a [`divmod`](https://www.w3schools.com/python/ref_func_divmod.asp) function that will produce both the quotient and the remainder of two numbers.